### ISSUE TITLE
Fails to initialize if unneeded profile types fail to load

### ISSUE THREAD

**Issue by ghost** [zowe/vscode-extension-for-zowe#796]
Current Behavior:
If I have an invalid Zowe CLI profile of any type (CICS, SSH, etc), Zowe Explorer will fail to initialize because that profile fails to load.

Expected Behavior:
Zowe Explorer should only fail to initialize if the profile types it needs to load are invalid. The profile types it needs are z/OSMF, and any profile types requested by the extender API. If other profile types fail to load, that failure should be ignored.

**Comment by crawr** [comment]
Hi @tjohnsonBCM , not sure if this is related to #742 ?

**Comment by t1m0thyj** [comment]
@crawr I don't believe it's related, as that issue has been closed but the problem I reported here still exists.

To reproduce this, I can make a malicious edit to a "cics" or "zftp" profile by adding the line `invalid YAML` at the top so that it can't be parsed as a valid YAML file.

I can still run a CLI command like `zowe zos-files download data-set` without any error, because it only loads "zosmf" profiles.

But when I open VS Code, Zowe Explorer fails with an error:
![image](https://user-images.githubusercontent.com/22344007/122274914-19a97000-ceb1-11eb-8403-0ece6d9d06a9.png)

The expected behavior would be that it only reads profiles from disk of type "zosmf", and is unaffected by syntax errors in YAML files for other profile types.
